node_settled_handler.go

```diff
@@ -143,24 +173,43 @@ func (p *NodeSettledHandler) handleFinalHtlc(ctx context.Context,
 
 	settledHash, err := p.persister.MarkHtlcSettled(ctx, htlcKey)
 	switch {
-	case err == persistence.ErrHtlcAlreadySettled:
-		return nil
+	case err == persistence.ErrHtlcAlreadyInStatus:
+		return nil, nil
 
 	// If the htlc is not found, the final resolution was for an htlc that isn't
 	// managed by lnmux.
 	case err == types.ErrHtlcNotFound:
-		return nil
+		return nil, nil
 
 	case err != nil:
-		return fmt.Errorf("unable to mark htlc %v settled: %w", key, err)
+		return nil, fmt.Errorf("unable to mark htlc %v settled: %w", key, err)
 	}
 
-	p.logger.Infow("Htlc final settled received",
-		"chanID", key.ChanID, "htlcID", key.HtlcID, "hash", settledHash)
+	return settledHash, nil
+}
+
+func (p *NodeSettledHandler) handleFinalFailedHtlc(ctx context.Context,
+	key types.CircuitKey) (*lntypes.Hash, error) {
 
-	if settledHash != nil {
-		p.settledCallback(*settledHash)
+	htlcKey := types.HtlcKey{
+		ChanID: key.ChanID,
+		HtlcID: key.HtlcID,
+		Node:   p.lnd.PubKey(),
 	}
 
-	return nil
+	failedHash, err := p.persister.MarkHtlcFailed(ctx, htlcKey)
```

Comment:
Instead of this duplication, maybe you can do a `persister.MarkHtlcFinal(ctx, htlckey, settled)`?